web/src/pages/index.tsx

```diff
@@ -271,306 +53,91 @@ const Home: NextPage = () => {
-                      base_count: response.base_count
-                    }};
-                  });
-
-                  scroll30();
-                  break;
-
-                case "done":
-
-                  // append the response to the entries, reset to normal
-                  setLoadState((s) => {
-                    if (s.state === "streaming") {
-                      setEntries([...new_entries, s.response]);
-                      setRunningIndex((i) => (i + ProcessText(s.response.content, 0)[1].size));
-                    }
-
-                    return {state: "idle"};
-                  });
-
-                  // add any potential followup questions
-                  var i = 0;
-                  while ('followup_' + i in data) {
-                    followups = [...followups, data['followup_' + i]];
-                    i++;
-                  }
-
-                  break read;
-
-                case "error":
-                  setEntries([...new_entries, {role: "error", content: data.error}]);
-                  setLoadState({state: "idle"});
-                  break read;
-
-              }
-            }
-            message = "";
-          }
-        }
-      }
-
-      enable(followups);
-      scroll30();
-
-    } else {
-    // ----------------- HUMAN AUTHORED CONTENT RETRIEVAL ------------------
-      const query_id = query.split("\n", 2)[0];
-
-      const res = await fetch(`${STAMPY_URL}/questions/${query_id}`, {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-          "Accept": "application/json",
-          "Allow-Control-Allow-Origin": "*"
-        },
-      });
-
-      if (!res.ok) {
-        enable([]);
-        setLoadState({state: "idle"});
-        setEntries([...new_entries, {role: "error", content: "POST Error: " + res.status}]);
-        return;
-      }
-
-      const data = (await res.json()).data;
-
-      setEntries([...new_entries, {
-        role: "stampy",
-        content: data.text,
-        url: "https://aisafety.info/?state=" + data.pageid,
-      }]);
-
-      // re-enable the searchbox, with the question that was just answered
-      // removed from the list of possible followups.
-
-      // create an array of new followup questions from the data
-      const f_new = data.relatedQuestions.map((f: any) => { return {
-        pageid: f.pageid!,
-        text: f.title!,
-        score: 0
-      };});
-
-      const fpids = new Set(f_new.map((f: Followup) => f.pageid));
-
-      enable((f_old: Followup[]) => {
-        const f_old_filtered = f_old.filter((f) => f.pageid !== data.pageid && !fpids.has(f.pageid));
-        return [...f_new, ...f_old_filtered].slice(0, MAX_FOLLOWUPS); // this is correct, it's N and not N-1 in javascript fsr
-      });
-
-      scroll30();
+      setRunningIndex(runningIndex + ProcessText(result.content, 0)[1].size);
     }
+    setEntries((prev) => [...prev, result]);
+    enable(followups || []);
```

Comment:
followups can be either a list of followups, or a function to update the current value, so both flows will work